xdem/coreg/base.py

```diff
@@ -379,7 +379,252 @@ def _preprocess_coreg_raster_input(
+    if inlier_mask is not None:
+        if isinstance(inlier_mask, Mask):
+            inlier_mask = inlier_mask.data.filled(False).squeeze()
+        else:
+            inlier_mask = np.asarray(inlier_mask).squeeze()
+            assert inlier_mask.dtype == bool, f"Invalid mask dtype: '{inlier_mask.dtype}'. Expected 'bool'"
+
+        if np.all(~inlier_mask):
+            raise ValueError("'inlier_mask' had no inliers.")
+    else:
+        inlier_mask = np.ones(np.shape(rst_elev), dtype=bool)
+
+    # TODO: Convert to point cloud?
+    # Convert geodataframe to vector
+    point_elev = point_elev.to_crs(crs=crs)
+
+    return rst_elev, point_elev, inlier_mask, transform, crs
+
+
+def _preprocess_coreg_fit_point_point(
+    reference_elev: gpd.GeoDataFrame, to_be_aligned_elev: gpd.GeoDataFrame
+) -> tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """Pre-processing and checks of fit for point-point input."""
+
+    ref_dem = reference_elev
+    tba_dem = to_be_aligned_elev.to_crs(crs=reference_elev.crs)
+
+    return ref_dem, tba_dem
+
+
+def _preprocess_coreg_fit(
+    reference_elev: NDArrayf | MArrayf | RasterType | gpd.GeoDataFrame,
+    to_be_aligned_elev: NDArrayf | MArrayf | RasterType | gpd.GeoDataFrame,
+    inlier_mask: NDArrayb | Mask | None = None,
+    transform: rio.transform.Affine | None = None,
+    crs: rio.crs.CRS | None = None,
+) -> tuple[
+    NDArrayf | gpd.GeoDataFrame, NDArrayf | gpd.GeoDataFrame, NDArrayb | None, affine.Affine | None, rio.crs.CRS | None
+]:
+    """Pre-processing and checks of fit for any input."""
+
+    if not all(
+        isinstance(elev, (np.ndarray, gu.Raster, gpd.GeoDataFrame)) for elev in (reference_elev, to_be_aligned_elev)
+    ):
+        raise ValueError("Input elevation data should be a raster, an array or a geodataframe.")
+
+    # If both inputs are raster or arrays, reprojection on the same grid is needed for raster-raster methods
+    if all(isinstance(elev, (np.ndarray, gu.Raster)) for elev in (reference_elev, to_be_aligned_elev)):
+        ref_elev, tba_elev, inlier_mask, transform, crs = _preprocess_coreg_fit_raster_raster(
+            reference_dem=reference_elev,
+            dem_to_be_aligned=to_be_aligned_elev,
+            inlier_mask=inlier_mask,
+            transform=transform,
+            crs=crs,
+        )
+
+    # If one input is raster, and the other is point, we reproject the point data to the same CRS and extract arrays
+    elif any(isinstance(dem, (np.ndarray, gu.Raster)) for dem in (reference_elev, to_be_aligned_elev)):
+        if isinstance(reference_elev, (np.ndarray, gu.Raster)):
+            raster_elev = reference_elev
+            point_elev = to_be_aligned_elev
+            ref = "raster"
+        else:
+            raster_elev = to_be_aligned_elev
+            point_elev = reference_elev
+            ref = "point"
+
+        rst_elev, point_elev, inlier_mask, transform, crs = _preprocess_coreg_fit_raster_point(
```

Comment:
here you're overwriting all variables, expect "raster_elev" that is replaced by "rst_elev". Maybe use the same name?

Reply:
Done